Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -549,7 +549,12 @@ namespace Babylon
                 const spirv_cross::ShaderResources resources = compiler.get_shader_resources();
                 assert(resources.uniform_buffers.size() == 1);
                 const spirv_cross::Resource& uniformBuffer = resources.uniform_buffers[0];
-                const spirv_cross::SmallVector<spirv_cross::Resource>& samplers = resources.separate_samplers;
+#if (BGFX_CONFIG_RENDERER_METAL)
```

Comment:
Is this define guaranteed to be available here? (i.e., is this an implementation detail of bgfx.cmake, or part of its public contract?) Could we instead consider embedding this behavior into a helper function inside `ShaderCompiler.h`, with implementations swapped out per platform by our existing mechanism?

Reply:
tbh, I find this change pretty ugly but it works and I didn't find a better solution yet. I hope to get it removed once work on validation tests begins. If no solution is found then it will be hidden in the platform shader cpp.